frame/bounties/src/lib.rs

```diff
@@ -653,12 +779,622 @@ decl_module! {
+
+			// Mutate Subbounty
+			SubBounties::<T>::try_mutate_exists(bounty_id, subbounty_id,
+				|maybe_subbounty| -> DispatchResult {
+
+				let mut subbounty = maybe_subbounty
+					.as_mut()
+					.ok_or(Error::<T>::InvalidIndex)?;
+
+				// Ensure subbounty is in expected state
+				if let SubBountyStatus::SubCuratorProposed { ref subcurator } = subbounty.status
+				{
+					ensure!(signer == *subcurator, Error::<T>::RequireSubCurator);
+
+					// Reserve subcurator deposit
+					let deposit = T::BountyCuratorDeposit::get() * subbounty.fee;
+					T::Currency::reserve(subcurator, deposit)?;
+					subbounty.curator_deposit = deposit;
+
+					subbounty.status = SubBountyStatus::Active {
+						subcurator: subcurator.clone(),
+					};
+					Ok(())
+				} else {
+					Err(Error::<T>::UnexpectedStatus.into())
+				}
+			})?;
+		}
+
+		/// Unassign subcurator from a subbounty.
+		///
+		/// The dispatch origin for this call can be
+		/// either `RejectOrigin` or any signed origin.
+		///
+		/// For the origin other than T::RejectOrigin,
+		/// Parent bounty must be in active state,
+		/// for this subbounty call to work. For origin
+		/// T::RejectOrigin execution is forced by ignoring
+		/// the state of parent bounty.
+		///
+		/// If this function is called by the `RejectOrigin` or
+		/// the master curator, we assume that the subcurator is
+		/// malicious or inactive.
+		///
+		/// As a result, subcurator deposit may be slashed.
+		///
+		/// If the origin is the subcurator, we take this as a sign they are
+		/// unable to do their job, and they willingly give up.
+		/// We could slash the deposit, but for now we allow them to
+		/// unreserve their deposit and exit without issue.
+		/// (We may want to change this if it is abused.)
+		///
+		/// Finally, the origin can be anyone if and only if the subcurator
+		/// is "inactive". Expiry update due of parent bounty is
+		/// used to estimate mature or inactive state of subcurator.
+		///
+		/// This allows anyone in the community to call out
+		/// that a subcurator is not doing their due diligence, and
+		/// we should pick a new subcurator. In this case the subcurator
+		/// deposit is slashed.
+		///
+		/// State of subbounty is moved to Added state
+		/// on successful call completion.
+		///
+		/// - `bounty_id`: ID pair Bounty ID.
+		/// - `subbounty_id`: ID pair SubBounty ID to cancel.
+		#[weight = <T as Config>::WeightInfo::unassign_subcurator()]
+		fn unassign_subcurator(
+			origin,
+			#[compact] bounty_id: BountyIndex,
+			#[compact] subbounty_id: BountyIndex,
+		) {
+			let maybe_sender = ensure_signed(origin.clone())
+				.map(Some)
+				.or_else(|_| T::RejectOrigin::ensure_origin(origin).map(|_| None))?;
+
+			// Ensure parent bounty exist & active, get active status info
+			let (master_curator, update_due) = Self::ensure_bounty_exist(
+				bounty_id,
+				maybe_sender.is_none(),
+			).map(|rval|
+				rval.unwrap_or((T::AccountId::default(), Zero::zero())),
```

Comment:
Sure, function `unassign_subcurator()` refactored to avoid usage of default account & update_due. Please let be know if implementation is deviated from expectation.